src/ssl.c

```diff
@@ -23183,18 +23340,32 @@ WOLFSSL_SESSION* wolfSSL_SESSION_dup(WOLFSSL_SESSION* session)
     }
     return copy;
 #else
-    WOLFSSL_MSG("wolfSSL_SESSION_dup was called "
-                "but HAVE_EXT_CACHE is not defined");
+    WOLFSSL_MSG("wolfSSL_SESSION_dup feature not compiled in");
     (void)session;
     return NULL;
 #endif /* HAVE_EXT_CACHE */
 }
 
-void FreeSession(WOLFSSL_SESSION* session, int isAlloced)
+#endif /* OPENSSL_EXTRA || HAVE_EXT_CACHE */
+
+void FreeSession(WOLFSSL_SESSION* session, void* heap)
 {
     if (session == NULL)
         return;
 
+#ifdef WOLFSSL_STATIC_MEMORY
+    if (heap == NULL)
```

Comment:
Why does the heap need to be passed in?
Doesn't the session have the heap pointer kept against it?